aten/src/ATen/native/ReduceOpsUtils.h

```diff
@@ -260,4 +257,47 @@ static TensorIterator make_reduction(
   return make_reduction(name, result1, result2, self, dim, keepdim, dtype, dtype);
 }
 
+static void zero_numel_check_dims(const Tensor& self, const int64_t dim, const char *fn_name) {
+  if (self.ndimension() == 0) {
+    TORCH_CHECK_INDEX(dim == 0 || dim == -1, fn_name,
+      ": Expected reduction dim -1 or 0 for scalar but got ", dim);
```

Comment:
"for scalar" -> "for a zero dimensional input" ?

Reply:
IMHO the colloqial term for `zero dimension input` is `scalar` and would serve to create more confusion than reduce it. If you ask me, the word "scalar" is much more descriptive than "zero dimensional input". WDYT?

Reply:
I would think a scalar's a number, like 5; would "for a zero-dimensional tensor input" be clearer?